server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryParseRequest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportRequest;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: request to extensions to parse context
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryParseRequest extends TransportRequest {
+
+    private String categoryClassName;
+    private byte[] context;
+
+    public NamedWriteableRegistryParseRequest(String categoryClassName, byte[] context) {
+        this.categoryClassName = categoryClassName;
+        this.context = context;
```

Comment:
The `byte[]` here is passed by reference, so the calling code still has access to the `context` object and could edit the internal values.  You should use `this.context = Arrays.copyOf(context, context.length);`

Reply:
Great suggestion, I'll definitely add this